umich_api/api_utils.py

```diff
@@ -47,31 +51,39 @@ def __init__(self, base_url, client_id, client_secret, client_scope, api_json =
         with open(api_json, encoding='utf-8') as api_file:
             apis = json.loads(api_file.read())
 
-        if client_scope in apis: 
-            api = apis.get(client_scope)
-            self.token_url = api.get('token_url')
-            self.api_call = sleep_and_retry(limits(calls=api.get('limits_calls'), period=api.get('limits_period'))(self._api_call))
-            self.access_token = self.get_access_token(self.token_url)
-        else: 
-            raise Exception(f"Scope {client_scope} not in known API dict")
+        self.scopes = defaultdict(dict)
+        # Setup all of the calls to the apis with the limits
+        for (client_scope, api) in viewitems(apis): 
+            self.scopes[client_scope]["api_call"] = sleep_and_retry(limits(calls=api.get('limits_calls'), period=api.get('limits_period'))(self._api_call))
+            # Each client_scope seems to need it's own access token. Just get them all here first
+            if "access_token" not in self.scopes[client_scope]:
+                self.scopes[client_scope]["access_token"] = self.get_access_token(api.get('token_url'), client_scope)
 
-    def _api_call(self, api_call, method="GET", payload=None):
-        # type: (str, str, Dict[str, str]) -> requests.Response
+
+    def api_call(self, api_call, client_scope, method="GET", payload=None):
         """Calls the specified API with the optional method
         
         :param api_call: API to call on this application
         :type api_call: str
+        :param client_scope: Client scope, must be present in the api_json file
+        :type client_scope: str
         :param method: Which HTTP method to use, defaults to "GET", optional
         :type method: str
         :param payload: Payload for POST/GET methods, optional
         :type payload: Dict[str, str]  
         :returns Requests object with response from API (or an Exception)
         :raises Exception: No Access Token
         """
+        return self.scopes[client_scope]["api_call"](api_call, self.scopes[client_scope]["access_token"], method, payload)
+
+    def _api_call(self, api_call, access_token, method="GET", payload=None):
+        # type: (str, str, Dict[str, str]) -> requests.Response
+        """ Internal method for making api calls 
+        """
 
         headers = {
             "accept" : "application/json",
```

Comment:
in the `api_utils.py` file the below statements should say `headers` instead of `header'

      
        elif method == "POST":
            resp = requests.post(api_url, headers=header, data=payload)
        elif method == "PUT":
            resp = requests.put(api_url, headers=header, data=payload)
```